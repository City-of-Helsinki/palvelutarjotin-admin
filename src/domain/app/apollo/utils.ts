import apolloClient from './apolloClient';

export const clearApolloCache = async () => {
  apolloClient.stop();
  await Promise.all([apolloClient.resetStore(), apolloClient.clearStore()]);
};
