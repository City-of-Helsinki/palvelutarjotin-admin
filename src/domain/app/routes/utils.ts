import { ROUTES } from './constants';

export const getCmsPath = (slug?: string | null) => {
  return ROUTES.CMS_PAGE.replace('/:slug', slug ?? '');
};
