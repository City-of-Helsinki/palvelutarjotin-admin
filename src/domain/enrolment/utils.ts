import { EnrolmentFieldsFragment } from '../../generated/graphql';

export const getEnrolmentFields = (
  enrolment: EnrolmentFieldsFragment | undefined | null
) =>
  enrolment
    ? {
        extraNeeds: enrolment.studyGroup.extraNeeds,
        language:
          enrolment.person?.language || enrolment.studyGroup.person.language,
        personInfo: [
          enrolment.person?.phoneNumber,
          enrolment.person?.emailAddress,
        ].filter((e) => e),
        studyGroupPersonInfo: [
          enrolment.studyGroup.person.phoneNumber,
          enrolment.studyGroup.person.emailAddress,
        ].filter((e) => e),
      }
    : {};
