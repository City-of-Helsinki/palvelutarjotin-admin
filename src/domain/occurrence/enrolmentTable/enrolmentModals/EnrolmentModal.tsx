import classNames from 'classnames';
import { Button, Checkbox, IconCross, Notification, TextArea } from 'hds-react';
import React, { ReactNode } from 'react';
import { useTranslation } from 'react-i18next';
import ReactModal from 'react-modal';

import EnrolleesList from './EnrolleesList';
import styles from './enrolmentModals.module.scss';

export type EnrolleeProps = {
  personName?: string;
  studyGroupName?: string | null;
  studyLevel?: string | null;
  groupSize?: number | null;
  amountOfAdult?: number | null;
};

interface EnrolmentModalProps {
  onClose: () => void;
  title: string;
  appElement?: HTMLElement;
  enrollees?: EnrolleeProps[];
  noteText?: string | ReactNode;
  noteType?: 'success' | 'decline';
  noteSection?: boolean;
  submitButtonText: string;
  handleSubmit: () => void;
  preview?: ReactNode | null;
  handlePreviewClick?: () => void;
  onMessageTextChange?: (message: string) => void;
  showPreviewButton?: boolean;
  messageText?: string;
  submitting?: boolean;
}

const EnrolmentModal: React.FC<EnrolmentModalProps> = ({
  children,
  title,
  appElement,
  enrollees,
  noteText,
  noteType = 'success',
  noteSection,
  submitButtonText,
  preview,
  messageText,
  showPreviewButton,
  submitting = false,
  handleSubmit,
  handlePreviewClick,
  onMessageTextChange,
  onClose,
}) => {
  const { t } = useTranslation();
  const [showMessageTextArea, setShowMessageTextArea] = React.useState(false);
  const hasEnrollees = enrollees && enrollees.length > 0;

  const renderModalContent = () => {
    return (
      <>
        {noteText && (
          <div
            className={classNames({
              [styles.infoNoteSuccess]: noteType === 'success',
              [styles.infoNoteDecline]: noteType === 'decline',
            })}
          >
            {noteText}
          </div>
        )}
        {hasEnrollees && <EnrolleesList enrollees={enrollees} />}
        {noteSection && (
          <div className={styles.addNoteSection}>
            <Checkbox
              className={styles.noteCheckbox}
              onChange={() => setShowMessageTextArea(!showMessageTextArea)}
              checked={showMessageTextArea}
              id="add-note"
              label={t('enrolment.enrolmentModal.addMessage')}
            />
            {noteType === 'success' && (
              <div className={styles.formRow}>
                <Notification
                  label={t('eventOccurrenceForm.infoTitleAutoAcceptance')}
                >
                  {t('eventOccurrenceForm.infoContentAutoAcceptance')}
                </Notification>
              </div>
            )}
            {showMessageTextArea && (
              <div>
                <TextArea
                  className={styles.noteTextArea}
                  value={messageText}
                  id="note-text-area"
                  label={t('enrolment.enrolmentModal.messageToParticipants')}
                  onChange={(e) => onMessageTextChange?.(e.target.value)}
                />
              </div>
            )}
          </div>
        )}
        <div className={styles.buttonsContainer}>
          <Button variant="secondary" onClick={onClose}>
            {t('enrolment.enrolmentModal.cancelEnrolment')}
          </Button>
          <div className={styles.buttonsRight}>
            {/* TODO: preview functionality */}
            {showPreviewButton && (
              <Button
                iconLeft={<div></div>}
                variant="supplementary"
                onClick={handlePreviewClick}
              >
                {t('enrolment.enrolmentModal.preview')}
              </Button>
            )}
            <Button
              variant="primary"
              onClick={handleSubmit}
              disabled={submitting}
            >
              {submitButtonText}
            </Button>
          </div>
        </div>
      </>
    );
  };

  return (
    <ReactModal
      isOpen
      bodyOpenClassName={styles.bodyOpen}
      className={classNames(styles.modal)}
      overlayClassName={styles.overlay}
      appElement={appElement}
    >
      <div className={styles.titleRow}>
        <p>{title}</p>
        <button className={styles.closeButton} onClick={onClose}>
          <IconCross />
          {t('common.buttonClose')}
        </button>
      </div>
      {preview ? preview : renderModalContent()}
      {children}
    </ReactModal>
  );
};

export default EnrolmentModal;
