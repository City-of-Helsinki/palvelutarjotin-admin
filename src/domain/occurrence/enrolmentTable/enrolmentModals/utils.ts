import { EnrolmentTemplateContextQuery } from '../../../../generated/graphql';
import { Language } from '../../../../types';

export const getEnrolmentTemplateContextJSON = (
  templateContextData: EnrolmentTemplateContextQuery | undefined,
  message: string,
  locale: Language
) => {
  const enrolment = templateContextData?.enrolment;
  const event = templateContextData?.enrolment?.occurrence.linkedEvent;
  if (enrolment && event) {
    const { studyGroup, occurrence } = enrolment;
    return JSON.stringify({
      event: {
        name: {
          [locale]: event.name[locale],
        },
      },
      study_group: {
        name: studyGroup.unitName,
        person: {
          email_address: studyGroup.person.emailAddress,
        },
      },
      occurrence: {
        start_time: occurrence.startTime,
        p_event: {
          linked_event_id: occurrence.linkedEvent?.id,
        },
      },
      custom_message: message,
    });
  }
  return null;
};
