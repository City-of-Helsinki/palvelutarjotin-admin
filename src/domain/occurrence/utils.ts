import { LINKEDEVENTS_CONTENT_TYPE } from '../../constants';
import {
  EventFieldsFragment,
  Language,
  OccurrenceFieldsFragment,
  SeatType,
  useEventQuery,
} from '../../generated/graphql';
import getLinkedEventsInternalId from '../../utils/getLinkedEventsInternalId';
import omitTypenames from '../../utils/omitTypename';
import { VIRTUAL_EVENT_LOCATION_ID } from '../event/constants';
import { PUBLICATION_STATUS } from '../events/constants';
import { EnrolmentType } from './enrolmentInfoFormPart/EnrolmentInfoFormPart';
import {
  OccurrenceSectionFormFields,
  TimeAndLocationFormFields,
} from './types';
/**
 * Get payload to create/edit occurrence
 * @param {object} values
 * @return {object}
 */
export const getOccurrencePayload = ({
  values,
  pEventId,
  isVirtual,
}: {
  values: OccurrenceSectionFormFields;
  pEventId: string;
  isVirtual: boolean;
}) => {
  return {
    startTime: values.startTime,
    endTime: values.endTime,
    languages: values.languages.map((lang) => ({ id: lang as Language })),
    pEventId,
    placeId: isVirtual ? VIRTUAL_EVENT_LOCATION_ID : values.occurrenceLocation,
    amountOfSeats: Number(values.amountOfSeats) || 0,
    minGroupSize: Number(values.minGroupSize) || null,
    maxGroupSize: Number(values.maxGroupSize) || null,
    seatType: values.oneGroupFills
      ? SeatType.EnrolmentCount
      : SeatType.ChildrenCount,
  };
};

export const getEnrolmentType = (event: EventFieldsFragment): EnrolmentType => {
  if (event.pEvent.externalEnrolmentUrl) {
    return EnrolmentType.External;
  }
  if (event.pEvent.enrolmentStart) {
    return EnrolmentType.Internal;
  }
  return EnrolmentType.Unenrollable;
};

<<<<<<< HEAD
export const getFormEnrolmentType = (
  event: EventFieldsFragment
): EnrolmentType => {
  if (event.pEvent.externalEnrolmentUrl) {
    return EnrolmentType.External;
  }
  if (event.pEvent.enrolmentStart || !event.location?.id) {
    return EnrolmentType.Internal;
  }
  return EnrolmentType.Unenrollable;
};
=======
>>>>>>> cb70104 (feat(enrolment-type): events enrolment info differs on different enrolment type)

export const getEventQueryVariables = (id: string) => ({
  id,
  include: ['location', 'keywords', 'audience', 'in_language'],
});

// Reused query, helps with refetching
export const useBaseEventQuery: typeof useEventQuery = ({
  variables,
  ...options
} = {}) => {
  return useEventQuery({
    variables: getEventQueryVariables(variables?.id ?? ''),
    ...options,
  });
};

export const getOccurrenceFields = (
  occurrence: OccurrenceFieldsFragment | undefined | null
) => {
  return occurrence
    ? {
        languages:
          occurrence.languages.edges.map((edge) => edge?.node?.id ?? '') ?? [],
      }
    : {};
};

export const getEditEventPayload = ({
  formValues,
  event,
}: {
  formValues: TimeAndLocationFormFields;
  event: EventFieldsFragment;
}) => {
  const {
    autoAcceptance,
    enrolmentEndDays,
    location,
    enrolmentStart,
    isVirtual,
    neededOccurrences,
    externalEnrolmentUrl,
    enrolmentType,
  } = formValues;
  const eventData = omitTypenames(event);

  const pEventEnrolmentFields = {
    [EnrolmentType.Internal]: {
      enrolmentStart,
      enrolmentEndDays: Number(enrolmentEndDays) || 0,
      neededOccurrences: Number(neededOccurrences) || 0,
      autoAcceptance,
      externalEnrolmentUrl: null,
    },
    [EnrolmentType.External]: {
      enrolmentEndDays: null,
      enrolmentStart: null,
      neededOccurrences: 0,
      autoAcceptance: false,
      externalEnrolmentUrl,
    },
    [EnrolmentType.Unenrollable]: {
      enrolmentEndDays: null,
      enrolmentStart: null,
      neededOccurrences: 0,
      autoAcceptance: false,
      externalEnrolmentUrl: null,
    },
  };

  return {
    name: eventData.name,
    // start_date and offers are mandatory on LinkedEvents to use dummy data
    startTime: eventData.startTime || '',
    // endTime needed
    // eslint-disable-next-line max-len
    // see ticket: https://helsinkisolutionoffice.atlassian.net/secure/RapidBoard.jspa?rapidView=40&projectKey=PT&modal=detail&selectedIssue=PT-437&assignee=557058%3A7f7be94a-c144-45ca-950c-6091dd896255
    endTime: eventData.endTime,
    offers: eventData.offers,
    shortDescription: eventData.shortDescription,
    description: eventData.description,
    images: eventData.images.map((image) => ({ internalId: image.internalId })),
    infoUrl: eventData.infoUrl,
    audience: eventData.audience.map((keyword) => ({
      internalId: keyword.internalId,
    })),
    inLanguage: eventData.inLanguage.map((lang) => ({
      internalId: lang.internalId,
    })),
    keywords: eventData.keywords.map((keyword) => ({
      internalId: keyword.internalId,
    })),
    location: {
      internalId: getLinkedEventsInternalId(
        LINKEDEVENTS_CONTENT_TYPE.PLACE,
        // If event is virtual, we use location id for internet events
        isVirtual ? VIRTUAL_EVENT_LOCATION_ID : location
      ),
    },
    draft: eventData.publicationStatus === PUBLICATION_STATUS.DRAFT,
    organisationId: eventData.pEvent.organisation?.id || '',
    pEvent: {
      contactEmail: eventData.pEvent.contactEmail,
      contactPersonId: eventData.pEvent.contactPerson?.id,
      contactPhoneNumber: eventData.pEvent.contactPhoneNumber,
      mandatoryAdditionalInformation:
        eventData.pEvent.mandatoryAdditionalInformation,
      ...pEventEnrolmentFields[enrolmentType],
    },
  };
};
