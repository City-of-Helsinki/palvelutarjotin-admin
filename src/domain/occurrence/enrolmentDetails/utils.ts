import type { TFunction } from 'i18next';
import capitalize from 'lodash/capitalize';

import {
  EnrolmentFieldsFragment,
  NotificationType,
} from '../../../generated/graphql';
import { translateValue } from '../../../utils/translateUtils';

export const getNotificationInfoText = (
  enrolment: EnrolmentFieldsFragment,
  t: TFunction
) => {
  const language = translateValue(
    'common.languages.',
    enrolment.person?.language || 'fi',
    t
  );

  let notificationsText = '';
  const hasEmailNotifications = t('enrolmentForm.labelHasEmailNotification');
  const hasSmsNotifications = t('enrolmentForm.labelHasSmsNotification');

  if (enrolment.notificationType === NotificationType.Email) {
    notificationsText = hasEmailNotifications;
  } else if (enrolment.notificationType === NotificationType.Sms) {
    notificationsText = hasSmsNotifications;
  } else if (enrolment.notificationType === NotificationType.EmailSms) {
    notificationsText = `${hasSmsNotifications}, ${hasEmailNotifications}`;
  }
  return capitalize(
    `${notificationsText}, ${t('enrolmentForm.labelLanguage')}: ${language}`
  );
};
