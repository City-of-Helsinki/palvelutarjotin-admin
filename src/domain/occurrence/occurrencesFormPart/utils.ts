import { DataProxy } from '@apollo/client/cache';
import { isBefore } from 'date-fns';
import { TFunction } from 'i18next';
import orderBy from 'lodash/orderBy';
import uniqueId from 'lodash/uniqueId';

import { EVENT_LANGUAGES } from '../../../constants';
import {
  AddOccurrenceMutation,
  DeleteOccurrenceMutation,
  EventDocument,
  EventQuery,
  EventQueryVariables,
  LanguageNode,
  LanguageNodeConnection,
  LanguageNodeEdge,
  OccurrenceFieldsFragment,
  OccurrencesOccurrenceSeatTypeChoices,
} from '../../../generated/graphql';
import { pageInfoMock } from '../../../utils/mockDataUtils';
import sortFavorably from '../../../utils/sortFavorably';
import { parseDateTimeString } from '../../../utils/time/utils';
import { OccurrenceSectionFormFields } from '../types';
import { getDateFromDateAndTimeString, getPlaceId } from '../utils';

export const getOrderedLanguageOptions = (t: TFunction) => {
  const languagesOrder = sortFavorably(
    Object.values(EVENT_LANGUAGES).map((language) =>
      t(`common.languages.${language.toLowerCase()}`)
    ),
    [EVENT_LANGUAGES.FI, EVENT_LANGUAGES.SV, EVENT_LANGUAGES.EN].map(
      (language) => t(`common.languages.${language.toLowerCase()}`)
    )
  );

  return Object.values(EVENT_LANGUAGES)
    .map((language) => ({
      label: t(`common.languages.${language.toLowerCase()}`),
      value: language,
    }))
    .sort(
      (a, b) =>
        languagesOrder.indexOf(a.label) - languagesOrder.indexOf(b.label)
    );
};

// fakeLanguages is copy pasted here from mockDataUtils so we don't need to include faker in prod build.
const fakeLanguages = (
  languages?: Partial<LanguageNode>[]
): LanguageNodeConnection => {
  const fakeLanguage = (overrides?: Partial<LanguageNode>): LanguageNode => ({
    id: 'fi',
    name: 'Finnish',
    __typename: 'LanguageNode',
    ...overrides,
  });

  const fakeLanguageNodeEdge = (
    overrides?: Partial<LanguageNode>
  ): LanguageNodeEdge => ({
    cursor: '',
    node: fakeLanguage(overrides),
    __typename: 'LanguageNodeEdge',
  });

  return {
    edges: languages?.map((language) => fakeLanguageNodeEdge(language)) || [],
    pageInfo: pageInfoMock,
    __typename: 'LanguageNodeConnection',
  };
};

export const getOptimisticCreateOccurrenceResponse = ({
  values,
  isVirtual = false,
  isBookable = false,
}: {
  values: OccurrenceSectionFormFields;
  isVirtual?: boolean;
  isBookable?: boolean;
}): AddOccurrenceMutation => {
  return {
    __typename: 'Mutation',
    addOccurrence: {
      __typename: 'AddOccurrenceMutationPayload',
      occurrence: {
        __typename: 'OccurrenceNode',
        id: uniqueId(),
        amountOfSeats: Number(values.amountOfSeats) || 0,
        cancelled: false,
        startTime: getDateFromDateAndTimeString(
          values.startDate,
          values.startTime
        ).toISOString(),
        endTime: getDateFromDateAndTimeString(
          values.endDate || values.startDate,
          values.endTime
        ).toISOString(),
        languages: fakeLanguages(
          orderBy(values.languages, undefined, 'asc').map((lang) => ({
            name: lang,
            id: lang,
          }))
        ),
        placeId: getPlaceId({ values, isVirtual, isBookable }),
        remainingSeats: Number(values.amountOfSeats) || 0,
        seatType: values.oneGroupFills
          ? OccurrencesOccurrenceSeatTypeChoices.EnrolmentCount
          : OccurrencesOccurrenceSeatTypeChoices.ChildrenCount,
        seatsApproved: 0,
        seatsTaken: 0,
        maxGroupSize: Number(values.maxGroupSize) || null,
        minGroupSize: Number(values.minGroupSize) || null,
        pEvent: {
          __typename: 'PalvelutarjotinEventNode',
          id: uniqueId(),
        },
      },
    },
  };
};

export const getOptimisticDeleteOccurrenceResponse =
  (): DeleteOccurrenceMutation => {
    return {
      __typename: 'Mutation',
      deleteOccurrence: {
        __typename: 'DeleteOccurrenceMutationPayload',
        clientMutationId: null,
      },
    };
  };

export const addOccurrencesToCache = ({
  data,
  proxy,
  eventVariables,
}: {
  eventVariables: EventQueryVariables;
  proxy: DataProxy;
  data: AddOccurrenceMutation | null | undefined;
}) => {
  if (data?.addOccurrence) {
    const { addOccurrence } = data;
    const cachedEvent = proxy.readQuery({
      query: EventDocument,
      variables: eventVariables,
    }) as EventQuery;

    // Find index where to insert new occurrence in the cache array
    const index = cachedEvent.event?.pEvent.occurrences?.edges.findIndex(
      (o) => {
        const occurrenceStartTime = new Date(o?.node?.startTime);
        const startTime = new Date(data.addOccurrence?.occurrence?.startTime);
        return isBefore(startTime, occurrenceStartTime);
      }
    );

    const newOccurrences =
      index != null && index > -1
        ? [
            ...(cachedEvent.event?.pEvent.occurrences?.edges ?? []).slice(
              0,
              index
            ),
            {
              __typename: 'OccurrenceNodeEdge',
              node: addOccurrence.occurrence,
            },
            ...(cachedEvent.event?.pEvent.occurrences?.edges ?? []).slice(
              index
            ),
          ]
        : [
            ...(cachedEvent.event?.pEvent.occurrences?.edges ?? []),
            {
              __typename: 'OccurrenceNodeEdge',
              node: addOccurrence.occurrence,
            },
          ];

    proxy.writeQuery({
      query: EventDocument,
      data: {
        event: {
          ...cachedEvent.event,
          pEvent: {
            ...cachedEvent.event?.pEvent,
            occurrences: {
              ...cachedEvent.event?.pEvent.occurrences,
              edges: newOccurrences,
            },
          },
        },
      },
    });
  }
};

export const deleteOccurrenceFromCache = ({
  eventVariables,
  occurrenceId,
  proxy,
}: {
  eventVariables: EventQueryVariables;
  proxy: DataProxy;
  occurrenceId: string;
}) => {
  const cachedEvent = proxy.readQuery({
    query: EventDocument,
    variables: eventVariables,
  }) as EventQuery;
  proxy.writeQuery({
    query: EventDocument,
    data: {
      event: {
        ...cachedEvent.event,
        pEvent: {
          ...cachedEvent.event?.pEvent,
          occurrences: {
            ...cachedEvent.event?.pEvent.occurrences,
            edges: [
              // Filter out deleted occurrence
              ...(cachedEvent.event?.pEvent.occurrences?.edges ?? []).filter(
                (e) => e?.node?.id !== occurrenceId
              ),
            ],
          },
        },
      },
    },
  });
};

export const getOccurrencerWithSameDateAlreadyExists = (
  formValues: OccurrenceSectionFormFields,
  occurrences: OccurrenceFieldsFragment[]
) => {
  return occurrences.some((occurrence) => {
    const formStartDateTime = parseDateTimeString(
      `${formValues.startDate} ${formValues.startTime}`
    );
    // use endDate is it is filled
    const formEndDateTime = parseDateTimeString(
      `${formValues.endDate || formValues.startDate} ${formValues.endTime}`
    );
    const occurrenceStartTime = new Date(occurrence.startTime);
    const occurrenceEndTime = new Date(occurrence.endTime);

    return (
      formStartDateTime.getTime() === occurrenceStartTime.getTime() &&
      formEndDateTime.getTime() === occurrenceEndTime.getTime()
    );
  });
};
