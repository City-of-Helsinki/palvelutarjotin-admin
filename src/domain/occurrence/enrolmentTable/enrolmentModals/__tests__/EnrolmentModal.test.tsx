import userEvent from '@testing-library/user-event';
import * as React from 'react';

import { render, screen } from '../../../../../utils/testUtils';
import EnrolmentModal from '../EnrolmentModal';

it('matches snapshot', () => {
  const { baseElement } = render(
    <EnrolmentModal
      title="Title"
      onClose={jest.fn()}
      appElement={document.body}
      submitButtonText={''}
      handleSubmit={function (): void {
        throw new Error('Function not implemented.');
      }}
    />
  );

  expect(baseElement).toMatchSnapshot();
});

it('renders correctly and calls onClose handler when close button is clicked', async () => {
  const onCloseHandler = jest.fn();
  render(
    <EnrolmentModal
      title="Title"
      onClose={onCloseHandler}
      appElement={document.body}
      submitButtonText={''}
      handleSubmit={function (): void {
        throw new Error('Function not implemented.');
      }}
    />
  );

  expect(screen.getByText(/Title/i)).toBeInTheDocument();

  const closeButton = screen.getByText(/sulje/i);
  await userEvent.click(closeButton);

  expect(onCloseHandler).toHaveBeenCalledTimes(1);
});
