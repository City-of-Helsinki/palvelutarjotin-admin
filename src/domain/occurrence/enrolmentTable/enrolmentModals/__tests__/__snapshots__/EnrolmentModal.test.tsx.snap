// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<body
  aria-hidden="true"
  class="bodyOpen"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay ReactModal__Overlay--after-open overlay"
    >
      <div
        class="ReactModal__Content ReactModal__Content--after-open modal"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="titleRow"
        >
          <p>
            Title
          </p>
          <button
            class="closeButton"
          >
            <svg
              class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
              role="img"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fill-rule="evenodd"
              >
                <rect
                  height="24"
                  width="24"
                />
                <polygon
                  fill="currentColor"
                  points="18 7.5 13.5 12 18 16.5 16.5 18 12 13.5 7.5 18 6 16.5 10.5 12 6 7.5 7.5 6 12 10.5 16.5 6"
                />
              </g>
            </svg>
            Sulje
          </button>
        </div>
        <div
          class="buttonsContainer"
        >
          <button
            class="Button-module_button__1msFE button_hds-button__2A0je Button-module_secondary__1nABp button_hds-button--secondary__1NOWS"
            type="button"
          >
            <span
              class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
            >
              Peruuta
            </span>
          </button>
          <div
            class="buttonsRight"
          >
            <button
              class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO"
              type="button"
            >
              <span
                class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
